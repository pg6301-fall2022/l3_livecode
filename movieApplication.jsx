import {Route, Routes, useNavigate} from "react-router-dom";
import * as React from "react";
import {useEffect, useState} from "react";

const MOVIES = [
    {
        title: "Plan 9 from outer space",
        year: 1957,
        plot: "utter confusion"
    },
    {
        title: "Lecture 2 - Web applications",
        year: 2022,
        plot: "also confusion, but more modern"
    },
]

function MovieCard({movie}){
    const {title, year, plot} = movie;

    return <div>
        <h2> {title} - ({year}) </h2>
        <p> {plot} </p>
    </div>;
}

export function ListMovies() {
    return <div>
        <h1> Movie List </h1>
        {MOVIES.map(movie => <MovieCard key={movie.title} movie={movie} />)}
    </div>;
}

export function MovieApplication() {
    return <Routes>
        <Route path={"/"} element={<ListMovies />}/>
        <Route path={"/new"} element={<CreateMovie />}/>
    </Routes>
}

function CreateMovie(){
    const [title, setTitle] = useState("");
    const [year, setYear] = useState("");
    const [plot, setPlot] = useState("");

    const [newMovie, setNewMovie] = useState({});

    const navigate = useNavigate();

    useEffect(() => {
        setNewMovie({title, year, plot});
    }, [title, year, plot]);

    function handleSubmit(event){
        event.preventDefault();
        MOVIES.push(newMovie);
        navigate("..");
    }

    return <form onSubmit={handleSubmit}>
        <h1> Create new movie </h1>
        <div>
            Title: <input value={title} onChange={event => setTitle(event.target.value)}/>
        </div>
        <div>
            Year: <input value={year} onChange={event => setYear(event.target.value)}/>
        </div>
        <div>
            Plot: <textarea value={plot} onChange={event => setPlot(event.target.value)}/>
        </div>
        <button>Submit</button>
        <pre>
            {JSON.stringify(newMovie)}
        </pre>
    </form>;
}


