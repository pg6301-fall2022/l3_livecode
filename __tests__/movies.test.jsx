import * as React from "react";
import {render} from "react-dom";
import {ListMovies} from "../movieApplication";

describe("movie pages", () => {

    it("shows movies list", () => {

        const element = document.createElement("div");
        render(<ListMovies />, element);
        expect(element.innerHTML).toMatchSnapshot();
    });

    it("test add new movies", () => {

    });

});