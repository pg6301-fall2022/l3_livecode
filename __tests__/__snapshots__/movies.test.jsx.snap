// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie pages shows movies list 1`] = `"<div><h1> Movie List </h1><div><h2> Plan 9 from outer space - (1957) </h2><p> utter confusion </p></div><div><h2> Lecture 2 - Web applications and API-design - (2022) </h2><p> also confusion, but more modern </p></div></div>"`;
